docs/features.md

```diff
@@ -1,58 +1,257 @@
 
-![Retrie Demo](https://i.imgur.com/Ev7B87k.gif)
+## Diagnostics
 
-### Explicit import lists
+### GHC compiler errors and warnings
+Provided by: `ghcide`
 
-![Imports code lens Demo](https://imgur.com/pX9kvY4.gif)
+Provides errors and warnings from GHC as diagnostics.
 
-### Wingman
+### Hlint hints
+Provided by: `hls-hlint-plugin`
 
-![Wingman Demo](https://user-images.githubusercontent.com/307223/92657198-3d4be400-f2a9-11ea-8ad3-f541c8eea891.gif)
+Provides hlint hints as diagnostics.
 
-### Hlint
+## Hovers
+Provided by: `ghcide`
 
-![Hlint Demo](https://user-images.githubusercontent.com/54035/110860028-8f9fa900-82bc-11eb-9fe5-6483d8bb95e6.gif)
+Type information and documentation on hover, [including from local definitions](./configuration.md#how-to-show-local-documentation-on-hover).
 
-### Module names
+## Jump to definition
+Provided by: `ghcide`
 
-![Module Name Demo](https://user-images.githubusercontent.com/54035/110860755-78ad8680-82bd-11eb-9845-9ea4b1cc1f76.gif)
+Jump to the definition of a name.
+
+Known limitations:
+- Only works for [local definitions](https://github.com/haskell/haskell-language-server/issues/708).
+
+## Jump to type definition
+Provided by: `ghcide`
+
+Known limitations:
+- Only works for [local definitions](https://github.com/haskell/haskell-language-server/issues/708).
+
+## Find references
+Provided by: `ghcide`
+
+Find references to a name within the project.
+
+## Completions
+
+### Code completions
+Provided by: `ghcide`
+
+- Completion of names from qualified imports.
+- Completion of names from non-imported modules.
+
+### Pragma completions
+Provided by: `hls-pragmas-plugin`
+
+Completions for language pragmas.
+
+## Formatting
+Format your code with various Haskell code formatters.
+
+| Formatter       | Provided by                  |
+|-----------------|------------------------------|
+| Brittany        | `hls-brittany-plugin`        |
+| Floskell        | `hls-floskell-plugin`        |
+| Fourmolu        | `hls-fourmolu-plugin`        |
+| Ormolu          | `hls-ormolu-plugin`          |
+| Stylish Haskell | `hls-stylish-haskell-plugin` |
+
+## Document symbols
+Provided by: `ghcide`
+
+Provides listing of the symbols defined in a module, used to power outline displays.
+
+## Workspace symbols
+Provided by: `ghcide`
 
-### Call hierarchy
+Provides listing of the symbols defined in the project, used to power searches.
+
+## Call hierarchy
+Provided by: `hls-call-hierarchy-plugin`
+
+Shows ingoing and outgoing calls for a function.
 
 ![Call Hierarchy in VSCode](https://github.com/haskell/haskell-language-server/raw/2857eeece0398e1cd4b2ffb6069b05c4d2308b39/plugins/hls-call-hierarchy-plugin/call-hierarchy-in-vscode.gif)
 
+## Highlight references
+Provided by: `ghcide`
+
+Highlights references to a name in a document.
+
+## Code actions
+
+### Insert missing pragmas
+Provided by: `hls-pragma-plugin`
+
+Code action kind: `quickfix`
+
+Inserts missing pragmas needed by GHC.
+
+### Apply Hlint fixes
```

Comment:
This mentioning then probably needs to mention the caveat, that during doing that (for sure during the "apply all hints" - it frequently deletes CPP code blocks across the whole file.

Reply:
Sorry, this may be out of date, because it appears I still use:
```
hlint --version
HLint v3.2.7
```
While there is `3.3.6`.

Reply:
No, this is an `apply-refact` limitation AFAICT, and good to include. I've added it.